### PR TITLE
Swap DataService.list to promise in membership

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Re issue #799 
Might as well get these rolling
@jwforres @spadgett

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
Rebased.

File: app/scripts/controllers/membership.js [hunk 1]
```diff
@@ -71,15 +71,16 @@ angular
...
+          .then(function(resp) {
             angular.extend($scope, {
               canShowRoles: true,
               roleBindings: resp.by('metadata.name'),
               subjectKindsForUI: MembershipService.mapRolebindingsForUI(resp.by('metadata.name'), allRoles)
             });
-            resetForm();
-          }, {
-            errorNotification: false
+			resetForm();
```
Comment (spadgett) [inline app/scripts/controllers/membership.js]: indentation

File: app/scripts/services/membership/roles.js [hunk 2]
```diff
@@ -4,37 +4,15 @@ angular
...
-        resolve(resp);
-      });
-
-      return deferred.promise;
+      return $q.all([
+        DataService
+          .list('roles', context, null),
+        DataService
+          .list('clusterroles', {}, null)
+      ]);
```
Comment (spadgett) [inline app/scripts/services/membership/roles.js]: 👍

File: app/scripts/controllers/membership.js (hunk 1 above)
Comment (benjaminapetersen) [inline app/scripts/controllers/membership.js]: nice, got it

**Comment by spadgett** [comment]
[merge]

**Pull request merged**